### PR TITLE
feat(*): move IPMI to a module

### PR DESCRIPTION
**Pull request by radireddy**
extract IPMI code as a reusable component into a separate module

ref #MANAGER-7282

Signed-off-by: Ravindra Adireddy <ravindra.adireddy@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/bm-components`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-7282
| License          | BSD 3-Clause

## Description

1. Extract IPMI code into bm-server-components as a reusable component
2. Use this component in dedicated server.
3. Cleanup the old code.

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/server/ipmi/controller.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  handleSuccess(message) {
+    this.Alerter.success(
+      message,
+      'server_dashboard_alert',
+    );
+  }
+
+  onKvmOrder() {
+    console.log('onKVMOrder');
```
Comment (frenautvh) [inline packages/manager/apps/dedicated/client/app/dedicated/server/ipmi/controller.js:22]:
```suggestion
```

File: packages/manager/apps/dedicated/client/app/css/source.less [hunk 2]
```diff
@@ -46,7 +46,7 @@
 @import 'less/dedicated/dedicatedcloud/datastores';
 @import 'less/dedicated/dedicatedcloud/security-options';
 @import 'less/dedicated/dedicatedcloud/users';
-@import 'less/dedicated/server/installation';
+// @import 'less/dedicated/server/installation';
```
Comment (frenautvh) [inline packages/manager/apps/dedicated/client/app/css/source.less:49]:
```suggestion
```

File: packages/manager/modules/bm-server-components/src/ipmi/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  .module(moduleName, [
+    'oui',
+    'pascalprecht.translate',
+    'ngTranslateAsyncLoader',
+    'ngAtInternet',
+    ovhManagerCore,
+  ])
+  .component('serverIpmi', component)
+  .service('IpmiService', service)
+  .service('Polling', polling)
```
Comment (frenautvh) [inline packages/manager/modules/bm-server-components/src/ipmi/index.js:25]: beware that polling is also used here for example :

https://github.com/ovh/manager/blob/master/packages/manager/apps/dedicated/client/app/dedicated/server/details/server.service.js#L27

if you only define the service in the lazy loaded ipmi module it won't be available for the rest of manager dedicated until that module is loaded

File: packages/manager/modules/bm-server-components/src/ipmi/ipmi.class.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+export default class BmServerComponentsIpmi {
+  constructor(server, features, urlkvm) {
```
Comment (frenautvh) [inline packages/manager/modules/bm-server-components/src/ipmi/ipmi.class.js:2]: missing ngInject

File: packages/manager/modules/bm-server-components/src/ipmi/index.js (hunk 3 above)
Comment (radireddy) [reply packages/manager/modules/bm-server-components/src/ipmi/index.js:25]: I have moved polling into this module and its been injected in dedicated app at root level.

**Comment by radireddy** [comment]
created a new PR https://github.com/ovh/manager/pull/5250

**Pull request closed without merging**